tests/compiler/array_without_size/ranges.leo

```diff
@@ -0,0 +1,10 @@
+/*
+namespace: Compile
+expectation: Pass
+input_file: input/dummy.in
+*/
+
+function main(a: [char; 11], y: bool) -> bool {
+    let z = a[1..2]; // open ranges are impossible
```

Comment:
Why that? You know the size of the array, so you can compute the true size of the range `1..` for example (`10` in this case).

Reply:
For the reasons stated in the comment that I've left for this PR. Currently we don't have a flattening stage which will substitute the unsized arrays' sizes.

Reply:
Could you extend the comment with an elaboration of what you wrote here ("Currently we don't have a flattening stage...")?

Reply:
Perhaps @damirka is referring to a case like
```
function main(a: [char; _], y: bool) -> bool {
    let z = a[1..];
    // ...
}
```
where the length of the array is not known until the code is flattened (a brief description of flattening is at https://github.com/AleoHQ/leo-reference/blob/main/src/introduction.md, under 'Zero-Knowledge Features'.

Reply:
That's different though; in that case the length of `a` is unknown. In the test case here, the type of `a` should be fully known and so the type of `a[1..]` would be fully known as well with some trivial math.